src/main/java/uk/gov/hmcts/ccd/domain/service/getevents/DefaultGetEventsOperation.java

```diff
@@ -45,12 +45,11 @@ public DefaultGetEventsOperation(CaseAuditEventRepository auditEventRepository,
             throw new BadRequestException("Case reference " + caseReference + " is not valid");
         }
 
-        final CaseDetails caseDetails =
-            getCaseOperation.execute(caseReference)
```

Comment:
@banderous just noticed a potential issue here. Before when retrieving a case the `getCaseOperation` was decorated with security decorators which checked whether had the rights to access the case. Now with the new impl that no longer the case. Are we losing authorization checks now?

Reply:
Good spot here, I have incorporated the authorisation checks into the query. Longer term there may be scope to refactor the way we access cases in a more composable way with fewer unique queries scattered across our code.